src/utils/ICS20Lib.sol

```diff
@@ -20,12 +20,25 @@ library ICS20Lib {
         string memo;
     }
 
-    /// @notice Convenience type used after unmarshalling the packet data and converting addresses
-    struct UnwrappedFungibleTokenPacketData {
-        address erc20ContractAddress;
-        uint256 amount;
+    /// @dev SendPacketData is a type used to represent the data needed to send a packet.
+    struct SendPacketData {
+        string denom;
+        bool receiverChainIsSource;
+        address erc20Contract;
         address sender;
         string receiver;
+        uint256 amount;
+        string memo;
+    }
+
+    /// @dev SendPacketData is a type used to represent the data needed to send a packet.
+    struct ReceivePacketData {
+        string denom;
+        bool senderChainIsSource;
+        address erc20Contract;
+        string sender;
+        address receiver;
+        uint256 amount;
         string memo;
     }
```

Comment:
We shouldn't introduce new structs. I am not convinced this is needed. We can add the extra bool as a return arguments to the functions that produce these structs.

Reply:
Yeah, the only real difference right now (except the bool, which you mentioned) is the address difference for sender and receiver. I can look into what it would look like to have a single struct and do the respective sender receiver unpacking in each method.